integration/querier_test.go

```diff
@@ -592,9 +593,33 @@ func TestQuerierWithChunksStorage(t *testing.T) {
 
 	wg.Wait()
 
+	c, err := e2ecortex.NewClient("", querier.HTTPEndpoint(), "", "", "user-0")
+	require.NoError(t, err)
+
+	// Ensure limit errors on ranged queries don't return 500s.
+	start := now.Add(-264 * time.Hour)
+	end := now
+	step := 264 * time.Hour
+
+	r, body, err := c.QueryRangeRaw("series_1", start, end, step)
+	require.NoError(t, err)
+	require.Equal(t, 422, r.StatusCode)
+	expected := `
+	{
+		"error":"expanding series: the query time range exceeds the limit (query length: 264h5m0s, limit: 240h0m0s)",
+		"errorType":"execution",
+		"status":"error"
+	}
+	`
+	require.JSONEq(t, expected, string(body))
+
 	// Ensure no service-specific metrics prefix is used by the wrong service.
 	assertServiceMetricsPrefixes(t, Distributor, distributor)
 	assertServiceMetricsPrefixes(t, Ingester, ingester)
 	assertServiceMetricsPrefixes(t, Querier, querier)
 	assertServiceMetricsPrefixes(t, TableManager, tableManager)
 }
+
+func formatTime(t time.Time) string {
```

Comment:
Shall we expose function from e2ecortex package instead?

Reply:
I think so.

Reply:
This was never meant to be here, I've included it in the e2e package already - is never used in this file.